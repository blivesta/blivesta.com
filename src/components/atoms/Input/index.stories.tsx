import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import Input from '.';

const Wrapper = styled.div`
  width: 480px;
  padding: 60px;
`;

storiesOf(`atoms|Input`, module).add(`default`, () => (
  <Wrapper>
    <Input type="text" placeholder="placeholder" />
  </Wrapper>
));
