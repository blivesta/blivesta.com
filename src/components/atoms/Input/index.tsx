import styled, { css } from 'styled-components';

export const inputStyle = css`
  width: 100%;
  padding: 0 ${({ theme }) => theme.spaces[1]};
  font-size: ${({ theme }) => theme.fontSizes[2]};
  color: ${({ theme }) => theme.colors.baseText[0]};
  background-color: ${({ theme }) => theme.colors.input.bg};
  border-top: 0;
  border-left: 0;
  border-right: 0;
  border-bottom: 1px solid ${({ theme }) => theme.colors.input.border};
  outline: none;
  box-shadow: none;
  transition: background, border-bottom ${({ theme }) => theme.transition};
  -webkit-appearance: none;
  border-radius: 0;

  &:focus {
    border-bottom: 1px solid ${({ theme }) => theme.colors.input.forcus.border};
    background-color: ${({ theme }) => theme.colors.input.forcus.bg};
  }
`;

const Input = styled.input`
  ${inputStyle}

  height: ${({ theme }) => theme.control.height};
`;

export default Input;
