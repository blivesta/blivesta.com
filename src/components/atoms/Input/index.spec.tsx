import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '.';

describe('Input', () => {
  it('snapshot', () => {
    const tree = shallow(<Input />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
