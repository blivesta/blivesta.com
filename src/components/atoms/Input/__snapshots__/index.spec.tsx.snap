// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  ",
          "
  width: 100%;
  font-size: ",
          [Function],
          ";
  color: ",
          [Function],
          ";
  background: transparent;
  outline: none;
  box-shadow: none;
  padding: 0 ",
          [Function],
          ";
  border-top: 0;
  border-left: 0;
  border-right: 0;
  border-bottom: 1px solid ",
          [Function],
          ";
  transition: background, border-bottom 0.2s;

  &:focus {
    border-bottom: 1px solid ",
          [Function],
          ";
    background: rgba(61, 16, 107, 0.15);
  }
",
          "

  height: 40px;
",
        ],
      },
      "displayName": "styled.input",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "input",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
